src/Std/Time/Zoned/ZonedDateTime.lean

```diff
@@ -0,0 +1,321 @@
+  Sigma.mk dt.fst (dt.snd.subMonthsClip months)
+
+/--
+Add `Month.Offset` to a `ZonedDateTime`, this function rolls over any excess days into the following
+month.
+-/
+def addMonthsRollOver (dt : ZonedDateTime) (months : Month.Offset) : ZonedDateTime :=
+  Sigma.mk dt.fst (dt.snd.addMonthsRollOver months)
+
+/--
+Add `Month.Offset` to a `ZonedDateTime`, this function rolls over any excess days into the following
+month.
+-/
+@[inline]
+def subMonthsRollOver (dt : ZonedDateTime) (months : Month.Offset) : ZonedDateTime :=
+  Sigma.mk dt.fst (dt.snd.subMonthsRollOver months)
+
+/--
+Add `Year.Offset` to a `ZonedDateTime`, this function rolls over any excess days into the following
+month.
+-/
+@[inline]
+def addYearsRollOver (dt : ZonedDateTime) (years : Year.Offset) : ZonedDateTime :=
+  Sigma.mk dt.fst (dt.snd.addYearsRollOver years)
+
+/--
+Add `Year.Offset` to a `ZonedDateTime`, it clips the day to the last valid day of that month.
+-/
+@[inline]
+def addYearsClip (dt : ZonedDateTime) (years : Year.Offset) : ZonedDateTime :=
+  Sigma.mk dt.fst (dt.snd.addYearsClip years)
+
+/--
+Add `Hour.Offset` to a `ZonedDateTime`, adjusting the date if necessary.
+-/
+def addHours (dt : ZonedDateTime) (hours : Hour.Offset) : ZonedDateTime :=
+  Sigma.mk dt.fst (dt.snd.addHours hours)
+
+/--
+Subtract `Hour.Offset` from a `ZonedDateTime`, adjusting the date if necessary.
+-/
+@[inline]
+def subHours (dt : ZonedDateTime) (hours : Hour.Offset) : ZonedDateTime :=
+  Sigma.mk dt.fst (dt.snd.subHours hours)
+
+/--
+Add `Minute.Offset` to a `ZonedDateTime`, adjusting the date if necessary.
+-/
+def addMinutes (dt : ZonedDateTime) (minutes : Minute.Offset) : ZonedDateTime :=
+  Sigma.mk dt.fst (dt.snd.addMinutes minutes)
+
+/--
+Subtract `Minute.Offset` from a `ZonedDateTime`, adjusting the date if necessary.
+-/
+@[inline]
+def subMinutes (dt : ZonedDateTime) (minutes : Minute.Offset) : ZonedDateTime :=
+  Sigma.mk dt.fst (dt.snd.subMinutes minutes)
+
+/--
+Add `Second.Offset` to a `ZonedDateTime`, adjusting the date if necessary.
+-/
+def addSeconds (dt : ZonedDateTime) (seconds : Second.Offset) : ZonedDateTime :=
+  Sigma.mk dt.fst (dt.snd.addSeconds seconds)
+
+/--
+Subtract `Second.Offset` from a `ZonedDateTime`, adjusting the date if necessary.
+-/
+@[inline]
+def subSeconds (dt : ZonedDateTime) (seconds : Second.Offset) : ZonedDateTime :=
+  Sigma.mk dt.fst (dt.snd.subSeconds seconds)
+
+/--
+Add `Nanosecond.Offset` to a `ZonedDateTime`, adjusting the date if necessary.
+-/
+def addNanoseconds (dt : ZonedDateTime) (nanoseconds : Nanosecond.Offset) : ZonedDateTime :=
+  Sigma.mk dt.fst (dt.snd.addNanoseconds nanoseconds)
+
+/--
+Subtract `Nanosecond.Offset` from a `ZonedDateTime`, adjusting the date if necessary.
+-/
+@[inline]
+def subNanoseconds (dt : ZonedDateTime) (nanoseconds : Nanosecond.Offset) : ZonedDateTime :=
+  Sigma.mk dt.fst (dt.snd.subNanoseconds nanoseconds)
+
+/--
+Determines the era of the given `PlainDate` based on its year.
```

Comment:
```suggestion
Determines the era of the given `ZonedDateTime` based on its year.
```
there are various other instances of the wrong type in docstrings, please check.